File: src/main/java/com/freya02/botcommands/api/modals/annotations/ModalData.java
package com.freya02.botcommands.api.modals.annotations;

import com.freya02.botcommands.api.modals.Modals;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Specifies that this parameter is supplied by the data sent in {@link Modals#create(String, String, Object...)}.
 * <br>The data supplied in the method above must be in the same order as the modal handler parameters, and the types must match.
 *
 * @see ModalHandler
 * @see ModalInput
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PARAMETER})
public @interface ModalData {}

File: BotCommands-core/src/main/kotlin/io/github/freya022/botcommands/api/modals/annotations/ModalData.kt
package io.github.freya022.botcommands.api.modals.annotations

import io.github.freya022.botcommands.api.modals.ModalBuilder

/**
 * Sets this parameter as data coming from [ModalBuilder.bindTo].
 *
 * The data supplied in the method above must be in the same order as the modal handler parameters,
 * and must match the types.
 *
 * @see ModalHandler @ModalHandler
 * @see ModalInput @ModalInput
 */
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class ModalData
